File: quasar/Quasar.DataTransfer/Produtos/Requests/ProdutoInserirRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Quasar.DataTransfer.Produtos.Requests
{
    public class ProdutoInserirRequest
    {
        public string Descricao { get; set; }
        public string Nome { get; set; }
        public decimal Valor { get; set; }
        public string Imagem { get; set; }
        public int CodigoCategoria { get; set; }
        public int CodigoFornecedor { get; set; }
        public int QuantidadeEstoque { get; set; }
        public EspecificacaoInserirRequest Especificacao { get; set; }
    }
}

File: quasar-front/src/app/produtos/models/requests/produtoInserir.request.ts
import { EspecificacaoInserirRequest } from "./especificacaoInserir.request";

export class ProdutoInserirRequest {
  public descricao: string;
  public nome: string;
  public valor: number;
  public imagem: string;
  public quantidadeEstoque: number;
  public codigoCategoria: number;
  public codigoFornecedor: number;
  public especificacao: EspecificacaoInserirRequest;

  constructor(params: Partial<ProdutoInserirRequest>) {
    this.descricao = params.descricao || '';
    this.nome = params.nome || '';
    this.valor = params.valor || 0;
    this.imagem = params.imagem || '';
    this.quantidadeEstoque = params.quantidadeEstoque || 0;
    this.codigoCategoria = params.codigoCategoria || 0;
    this.codigoFornecedor = params.codigoFornecedor || 0;
    this.especificacao = params.especificacao || new EspecificacaoInserirRequest({});
  }
}
